Changes needed to restore Payments coverage

AndreiH opened the pull request:
These changes are also from @bobsilverberg 's PR https://github.com/mozilla/marketplace-tests-gaia/pull/66
First adhoc run here: http://jenkins1.qa.scl3.mozilla.com/view/Marketplace/job/flame-kk-319.mozilla-b2g32_v2_0.ui.marketplace-tests.adhoc/22/console

AndreiH commented:
Today I fixed all remaining failures for Payments tests and xfailed two of the tests.
@bobsilverberg r?

AndreiH commented on marketplacetests/tests/test_marketplace_login_during_purchase.py:
```diff
@@ -26,7 +30,7 @@ def test_purchase_app(self):
         details_page = marketplace.navigate_to_app(APP_NAME)
         ff_accounts = details_page.tap_purchase_button(is_logged_in=False)
-        ff_accounts.login(acct.email, acct.password)
+        ff_accounts.login(username, password)
```
I changed this to the credentials user because the create/confirm window appears if we login with a new user

bobsilverberg commented:
Great work getting the payments tests working again @AndreiH!  Everything is passing for me locally, except for the two tests that are xfailed. Unfortunately one of them isn't failing in the same way I would expect from the xfail message.

`test_marketplace_login_during_purchase` is failing with the Install Confirmation dialog on the screen, as opposed to with the Buy dialog on the screen. Watching the test run, after the login is complete, it looks like the test returns to the Buy dialog, and then clicks on the Install button again. That takes us to the "Install Test Zippy With Me?" dialog and it is on that screen that the test times out. It is strange because I do not see any code that would make the phone click on that Install button again, so it seems like maybe the app is doing it automatically. Please run this a few times to see if you are seeing the same behaviour, and if so we should discuss this with @krupa to see if this is expected and how to best test this feature.

AndreiH commented:
I can see now that this is happening because the app was already bought in the past by this account and that's why we now see the "Install" button status instead of "Buy" or "Purchase" after log in.
It's supposed to automatically start the install or buy process because this is testing just that (logging in during purchase).
So, I changed the test back to creating a new account every time so that the button status changes to "Purchasing" after clicking on it, being a new account.
We are also not bothered by the "creating a pin" window that appears after the login because we then "Switch back to the Marketplace frame and wait for the install button to update" , meaning that the log in was successful and we are now in the purchasing step: https://github.com/AndreiH/marketplace-tests-gaia/blob/bob_changes_payments/marketplacetests/tests/test_marketplace_login_during_purchase.py#L32

I also changed the way we check for the button status (verifying the class attribute). Please see class attribute in the screenshots:
Before logging in:
![screenshot from 2015-01-07 16 15 59](https://cloud.githubusercontent.com/assets/2787766/5647544/8a0f91be-968f-11e4-8238-ab96f3378be5.png)

After logging in: 
![screenshot from 2015-01-07 16 17 18](https://cloud.githubusercontent.com/assets/2787766/5647549/936d0ad4-968f-11e4-9a2f-93f8e0c9520d.png)

AndreiH commented:
Closed it by mistake

AndreiH commented:
Adhoc run here: http://jenkins1.qa.scl3.mozilla.com/job/flame-kk-319.mozilla-b2g32_v2_0.ui.marketplace-tests.adhoc/27/HTML_Report/

bobsilverberg commented on marketplacetests/tests/test_marketplace_login_during_purchase.py:
```diff
@@ -31,4 +32,4 @@ def test_purchase_app(self):
         # Switch back to the Marketplace frame and wait for the install button to update
```
We should complete the create/confirm pin workflow which should return us to the expected screen.

AndreiH commented:
@bobsilverberg this has been updated and commits squashed

bobsilverberg commented:
Great work @AndreiH! The login during purchase test is working well for me. I'm going to do one final complete run of the tests locally and then I will merge this.

bobsilverberg commented:
This has been a long time coming, so I'm going to merge it, even though there was an issue unrelated to this PR with `test_marketplace_search_and_install_app`. I will open a separate issue for that.
